shared/actions/chat2/index.tsx

```diff
@@ -1794,23 +1795,12 @@ const messageSendByUsernames = async (
       },
       action.payload.waitingKey
     )
-    await RPCChatTypes.localPostTextNonblockRpcPromise(
-      {
-        body: action.payload.text.stringValue(),
-        clientPrev: Constants.getClientPrev(state, Types.conversationIDToKey(result.conv.info.id)),
-        conversationID: result.conv.info.id,
-        identifyBehavior: RPCTypes.TLFIdentifyBehavior.chatGui,
-        outboxID: null,
-        tlfName,
-        tlfPublic: false,
-      },
-      action.payload.waitingKey
-    )
-
-    // If there are block buttons on this conversation, clear them.
-    if (state.chat2.blockButtonsMap.has(result.conv.info.triple.tlfid.toString('hex'))) {
-      return Chat2Gen.createDismissBlockButtons({teamID: result.conv.info.triple.tlfid.toString('hex')})
-    }
+    const { text, waitingKey } = action.payload
+    await Saga.put(Chat2Gen.createMessageSend({
```

Comment:
pretty sure `Saga.put` will return immediately without error, this doesn't need to be async. Saga will schedule the calling of `messageSend`.

Reply:
this will not do anything. if you're using promises you return your final action

Reply:
the above creates a payload, wraps it in a put effect and waits for it to be created, then throws it away